tests/suites/test_suite_psa_crypto.function

```diff
@@ -294,6 +295,19 @@ exit:
     ((void) 0)
 #endif /* MBEDTLS_PSA_CRYPTO_SE_C */
 
+uint64_t parse_hex_string(data_t *hex_string)
+{
+    uint64_t result = 0;
+    for (size_t i = 0; i < hex_string->len; i++) {
+        if (MBEDTLS_IS_BIG_ENDIAN) {
```

Comment:
I don't think this should depend on the platform's endianness - otherwise we'll need different entries in the `.data` file for different platforms. We usually use big endian in our test data unless there's a reason not to.

Reply:
Note: this is also causing some (if not all) of the CI failures, as some compilers are complaining about dead code. Once this is removed, we'll be able to see if there are other failures in the CI.